sdk/core/azure-core-test/inc/azure/core/test/test_base.hpp

```diff
@@ -21,21 +30,163 @@ namespace Azure { namespace Core { namespace Test {
+        options.PerRetryPolicies.push_back(m_interceptor->GetRecordPolicy());
+      }
+    }
+
+    std::string Sanitize(std::string const& src)
+    {
+      std::string updated(src);
+      std::replace(updated.begin(), updated.end(), '/', '-');
+      return updated;
+    }
+
   protected:
     Azure::Core::Test::TestContextManager m_testContext;
     std::unique_ptr<Azure::Core::Test::InterceptorManager> m_interceptor;
 
+    // Reads the current test instance name.
+    // Name gets also sanitized (special chars are removed) to avoid issues when recording or
+    // creating
+    std::string GetTestName(bool sanitize = false)
+    {
+      std::string testName(::testing::UnitTest::GetInstance()->current_test_info()->name());
+      if (sanitize)
+      {
+        // replace `/` for `-`. Parameterized tests adds this char automatically to join the test
+        // name and the parameter suffix.
+        return Sanitize(testName);
+      }
+
+      return testName;
+    }
+
+    // Creates the sdk client for testing.
+    // The client will be set for record and playback before it is created.
+    Azure::Core::Credentials::TokenCredentialOptions GetTokenCredentialsOptions()
+    {
+      // Run instrumentation before creating the client
+      Azure::Core::Credentials::TokenCredentialOptions options;
+      PrepareOptions(options);
+      return options;
+    }
+
+    // Creates the sdk client for testing.
+    // The client will be set for record and playback before it is created.
+    template <class T, class O>
+    std::unique_ptr<T> InitTestClient(
+        std::string const& url,
+        std::shared_ptr<Core::Credentials::TokenCredential>* credential,
+        O& options)
+    {
+      // Run instrumentation before creating the client
+      PrepareClientOptions(&credential, options);
+      return std::make_unique<T>(url, *credential, options);
+    }
+
+    // Updates the time when test is on playback
+    void UpdateWaitingTime(std::chrono::milliseconds& current)
+    {
+      if (m_testContext.IsPlaybackMode())
+      {
+        current = 0ms;
+      }
+    }
+
+    // Util for tests to introduce delays
+    void TestSleep(std::chrono::milliseconds const& ms = 1s)
+    {
+      if (m_testContext.IsPlaybackMode())
+      {
+        return;
+      }
+      std::this_thread::sleep_for(ms);
+    }
+
+    void TestLog(std::string const& message)
+    {
+      using Azure::Core::Diagnostics::Logger;
+      using Azure::Core::Diagnostics::_internal::Log;
+      Log::Write(
+          Logger::Level::Verbose,
+          "Test Log from: [ " + m_testContext.GetTestPlaybackRecordingName() + " ] - " + message);
+    }
+
+    // Util for tests getting env vars
+    std::string GetEnv(const std::string& name)
+    {
+#if defined(_MSC_VER)
+#pragma warning(push)
+#pragma warning(disable : 4996)
+      const char* ret = std::getenv(name.data());
+#pragma warning(pop)
+#else
+      const char* ret = std::getenv(name.data());
+#endif
```

Comment:
```suggestion
#if defined(_MSC_VER)
#pragma warning(push)
// Warning C4996: This function or variable may be unsafe.
#pragma warning(disable : 4996)
#endif // _MSC_VER
      const char* const ret = std::getenv(name.data());
#if defined(_MSC_VER)
#pragma warning(pop)
#endif // _MSC_VER
```

Reply:
It is easier to read, when the code is the same, and we know we're using the same version of the code on all platforms, it is just compiler-specific warnings that we're taking care of.